### ISSUE TITLE
After upgrading to 3.6.0, OCI runtime exec error when trying to log in to pantheon

### ISSUE THREAD

**Issue by ptmkenny** [lando/pantheon#91]
Steps to reproduce

1. `lando pull`
2. "Enter a Pantheon machine token" (enter valid token)

`OCI runtime exec failed: exec failed: container_linux.go:380: starting container process caused: exec: "/helpers/pull.sh": stat /helpers/pull.sh: no such file or directory: unknown`

I tried `lando destroy`, `rm -rf ~/.lando`, and then `lando pull` again; same error.

This setup was working on lando `3.5.0`.

```
lando terminus -V
Terminus 3.0.2-dev
```

Perhaps it is related to the dev version of terminus?

**Comment by labboy0276** [comment]
Hi @ptmkenny we are running a patched version of terminus so php 8.0 can work with Pantheon / Lando.  There is one more issue they are working on until we use a regular tagged version of terminus.

All our tests pass, so I am curious why this is happening on your end.  What OS are you using?

**Comment by ptmkenny** [comment]
Thanks for the fast response.

I'm using lando through WSL2: Ubuntu 20.04 on Windows 11 Pro (10.0.22000).  This is a clean install of everything I set up about a week ago.

**Comment by labboy0276** [comment]
OK let me see if I can replicate @ptmkenny

**Comment by ptmkenny** [comment]
@labboy0276 Thanks.

Some more info: I get the same error even when I enter a garbage (invalid) Pantheon machine token.

After running `lando rebuild -y` a few more times, I also got a slightly different error at the same point (immediately after entering my machine token):

```
 lando pull
? Enter a Pantheon machine token [hidden]
Attempting to login via terminus...
PHP Warning:  scandir(/var/www/.terminus/cache/tokens/session): Failed to open directory: Not a directory in phar:///usr/local/bin/terminus/src/DataStore/FileStore.php on line 125

Warning: scandir(/var/www/.terminus/cache/tokens/session): Failed to open directory: Not a directory in phar:///usr/local/bin/terminus/src/DataStore/FileStore.php on line 125
PHP Warning:  scandir(/var/www/.terminus/cache/tokens/session): Failed to open directory: Not a directory in phar:///usr/local/bin/terminus/src/DataStore/FileStore.php on line 125

Warning: scandir(/var/www/.terminus/cache/tokens/session): Failed to open directory: Not a directory in phar:///usr/local/bin/terminus/src/DataStore/FileStore.php on line 125
PHP Warning:  scandir(/var/www/.terminus/cache/tokens/session): Failed to open directory: Not a directory in phar:///usr/local/bin/terminus/src/DataStore/FileStore.php on line 125

Warning: scandir(/var/www/.terminus/cache/tokens/session): Failed to open directory: Not a directory in phar:///usr/local/bin/terminus/src/DataStore/FileStore.php on line 125
anon@dc:~/d/anata$ 

```

**Comment by labboy0276** [comment]
@ptmkenny interesting, I hope to go to this soon.  It may be terminus after all since I am still pulling from source and something could of changed.  

For now, I would just rollback lando and I will get to this before the next patch release next week.

**Comment by labboy0276** [comment]
Actually @ptmkenny give me a hour, I think I know whats up now.

**Comment by ptmkenny** [comment]
Awesome!  No rush though-- I have a backup local db dump that I can do my work off of for now.

**Comment by labboy0276** [comment]
@ptmkenny do a lando destroy and rebuild.  See if your terminus is version is 3.0.3 now.

If so, it should clear up the issues.

**Comment by ptmkenny** [comment]
@labboy0276 Did `lando destroy -y; lando rebuild -y`.  `lando terminus -V` returns 3.0.3.

Then I tried to run `lando pull`; added machine token, got the same error:

```
Attempting to login via terminus...
PHP Warning:  scandir(/var/www/.terminus/cache/tokens/session): Failed to open directory: Not a directory in phar:///usr/local/bin/terminus/src/DataStore/FileStore.php on line 125

Warning: scandir(/var/www/.terminus/cache/tokens/session): Failed to open directory: Not a directory in phar:///usr/local/bin/terminus/src/DataStore/FileStore.php on line 125
PHP Warning:  scandir(/var/www/.terminus/cache/tokens/session): Failed to open directory: Not a directory in phar:///usr/local/bin/terminus/src/DataStore/FileStore.php on line 125

Warning: scandir(/var/www/.terminus/cache/tokens/session): Failed to open directory: Not a directory in phar:///usr/local/bin/terminus/src/DataStore/FileStore.php on line 125
PHP Warning:  scandir(/var/www/.terminus/cache/tokens/session): Failed to open directory: Not a directory in phar:///usr/local/bin/terminus/src/DataStore/FileStore.php on line 125

Warning: scandir(/var/www/.terminus/cache/tokens/session): Failed to open directory: Not a directory in phar:///usr/local/bin/terminus/src/DataStore/FileStore.php on line 125
```

Just to be sure, I went to the Pantheon dashboard, generated a new machine token, and tried that-- same error.

Next, I tried to log in with the machine token:

`lando terminus auth:login --machine-token=ACTUAL_MACHINE_TOKEN`

This produced the error:

```
PHP Warning:  scandir(/var/www/.terminus/cache/tokens/session): Failed to open directory: Not a directory in phar:///usr/local/bin/terminus/src/DataStore/FileStore.php on line 125

Warning: scandir(/var/www/.terminus/cache/tokens/session): Failed to open directory: Not a directory in phar:///usr/local/bin/terminus/src/DataStore/FileStore.php on line 125
```

**Comment by labboy0276** [comment]
Can you try purging all your docker containers and start over @ptmkenny

**Comment by labboy0276** [comment]
Actually, I am able to replicate this in tests now @ptmkenny https://github.com/lando/pantheon/runs/4570548101?check_suite_focus=true

Are you on 3.6.0 when you get this error?

**Comment by ptmkenny** [comment]
@labboy0276 Yes, I'm on 3.6.0.

Also, `.lando.yml` is:

```
config:
  env: lando
  xdebug: true
  framework: drupal8
  webroot: web
```

And `.pantheon.yml` is set to PHP 8, which is why I updated to 3.6.0.

I purged all containers (`lando poweroff; docker rm -vf $(docker ps -aq)`) and then ran `lando start; lando pull` but I'm still getting the same error.

**Comment by labboy0276** [comment]
@ptmkenny can you try doing this:

```
lando ssh -s appserver
mkdir -p /var/www/.terminus/cache/tokens/session
```

Then do a `lando restart` and try logging out and in and see if that lets you pull

**Comment by labboy0276** [comment]
I also had a couple people try and it seems fresh token fix the issue for them.

**Comment by ptmkenny** [comment]
@labboy0276 Thanks, I have it working now.

However, I ran into a problem.

When I tried 

`mkdir -p /var/www/.terminus/cache/tokens/session`, it told me it already existed, but when I ran:

 ```
www-data@0554ed1a06af:~/.terminus/cache/tokens$ cd /var/www/.terminus/cache/tokens/session
bash: cd: /var/www/.terminus/cache/tokens/session: Not a directory
```

So I deleted `session` and then re-created it as a directory and then pulled, and now everything is working.  Thanks! I'll leave this open for now if there is a bug fix that goes with it; otherwise, this can be considered closed.
